lms/djangoapps/bookmarks/views.py

```diff
@@ -166,6 +167,25 @@ def get_queryset(self):
 
         return api.get_bookmarks(user=self.request.user, course_key=course_key, serialized=False)
 
+    def paginate_queryset(self, queryset, page_size=None):
+        """ Override GenericAPIView.paginate_queryset for the purpose of eventing """
+        page = super(BookmarksListView, self).paginate_queryset(queryset, page_size)
+        course_id = self.request.QUERY_PARAMS.get('course_id')
+        event_data = {
+            'list_type': 'all_courses',
+            'bookmarks_count': page.paginator.count,
+            'page_size': self.get_paginate_by(),
+            'page_number': page.number,
+        }
+
+        if course_id is not None:
```

Comment:
You can move line 173 here. Also parse the `course_id`; only if it is valid should these changes be made to the `event_data`.